api/src/test/java/org/pmiops/workbench/workspaces/resources/UserRecentResourceServiceTest.java

```diff
@@ -180,46 +265,132 @@ public void testUserLimit() {
     }
 
     CLOCK.increment(2000);
-    long rowsCount = userRecentResourceDao.count();
+    long rowsCount = userRecentlyModifiedResourceDao.count();
     assertThat(rowsCount).isEqualTo(UserRecentResourceService.USER_ENTRY_COUNT);
 
     userRecentResourceService.updateNotebookEntry(
         newWorkspace.getWorkspaceId(),
         user.getUserId(),
         "gs://someDirectory/notebooks/notebookExtra");
-    rowsCount = userRecentResourceDao.count();
+    rowsCount = userRecentlyModifiedResourceDao.count();
     assertThat(rowsCount).isEqualTo(UserRecentResourceService.USER_ENTRY_COUNT);
-    DbUserRecentResource cache =
-        userRecentResourceDao.findByUserIdAndWorkspaceIdAndNotebookName(
+    DbUserRecentlyModifiedResource cache =
+        userRecentlyModifiedResourceDao.getResource(
             newWorkspace.getWorkspaceId(),
             user.getUserId(),
+            DbUserRecentlyModifiedResourceType.NOTEBOOK,
             "gs://someDirectory1/notebooks/notebook");
 
     assertThat(cache).isNull();
   }
 
-  //  We do test notebook deletion because it is a path reference
-  //  We do not test cohort or concept deletion because these are fk refs with
-  //  on delete cascade rule in place (no need to test db functionality)
   @Test
   public void testDeleteNotebookEntry() {
     userRecentResourceService.updateNotebookEntry(
         workspace.getWorkspaceId(), user.getUserId(), "gs://someDirectory1/notebooks/notebook1");
-    long rowsCount = userRecentResourceDao.count();
+    long rowsCount = userRecentlyModifiedResourceDao.count();
     assertThat(rowsCount).isEqualTo(1);
     userRecentResourceService.deleteNotebookEntry(
         workspace.getWorkspaceId(), user.getUserId(), "gs://someDirectory1/notebooks/notebook1");
-    rowsCount = userRecentResourceDao.count();
+    rowsCount = userRecentlyModifiedResourceDao.count();
     assertThat(rowsCount).isEqualTo(0);
   }
 
   @Test
   public void testDeleteNonExistentNotebookEntry() {
-    long rowsCount = userRecentResourceDao.count();
+    long rowsCount = userRecentlyModifiedResourceDao.count();
     assertThat(rowsCount).isEqualTo(0);
     userRecentResourceService.deleteNotebookEntry(
         workspace.getWorkspaceId(), user.getUserId(), "gs://someDirectory1/notebooks/notebook");
-    rowsCount = userRecentResourceDao.count();
+    rowsCount = userRecentlyModifiedResourceDao.count();
+    assertThat(rowsCount).isEqualTo(0);
+  }
+
+  @Test
+  public void testDeleteCohortEntry() {
+    userRecentResourceService.updateCohortEntry(
```

Comment:
When does this cohort get created? I see the test below (testDeleteNonExistentCohortEntry) seemingly only differing in the call to updateCohortEntry. Does updateCohortEntry create a cohort if it does not exist? I tried looking at the code for updateCohortEntry, and I do not see where that would be done.

Reply:
The Cohort is initialized in setUp().

The Cohort **Recent Resource Entry** is the difference between these two tests: one created a Recent Resource Entry and the other does not